src/scripts/main.js

```diff
@@ -1,3 +1,69 @@
-'use strict';
+const table = document.querySelector('.field');
+const addRowBtn = document.querySelector('.append-row');
+const removeRowBtn = document.querySelector('.remove-row');
+const addColBtn = document.querySelector('.append-column');
+const removeColBtn = document.querySelector('.remove-column');
 
-// write code here
+const maxLimit = 10;
+const minLimit = 2;
+
+function updateButtonStates() {
+  const rowCount = table.rows.length;
+  const colCount = table.rows[0].cells.length;
```

Comment:
Ensure that the table has at least one row before accessing `table.rows[0]`. If the table is initially empty, this line will throw an error. Consider adding a check to handle this case.